### PR TITLE
Bug 762978 - Make AndroidBrowserRepositorySession.recordToGuid map from ...

### PR DESCRIPTION
**Pull request by ncalexan**
...hashes of record strings to record guids to reduce memory pressure.

### CODE REVIEW

**Comment by indygreg** [comment]
I thought internally strings were represented as integers when stored in data structures. Or, is it just C# that makes this optimization?

**Comment by ncalexan** [comment]
It's possible, but I've not heard it.  (I'm hardly a JVM expert.)  Even if they are hashed in some way, the Strings must still be in memory, since the key set can be re-constructed.  Here the key set Strings take a lot of memory and I can inexpensively re-create any individual String, so storing only integers should reduce the amount of memory for storing the Map.

File: src/main/java/org/mozilla/gecko/sync/repositories/android/AndroidBrowserRepositorySession.java [hunk 1]
```diff
@@ -660,7 +695,7 @@ private void createRecordToGuidMap() throws NoGuidForIdException, NullCursorExce
         if (record != null) {
           final String recordString = buildRecordString(record);
           if (recordString != null) {
-            recordToGuid.put(recordString, record.guid);
+            recordToGuid.put(recordString.hashCode(), record.guid);
```
Comment (liuche) [inline src/main/java/org/mozilla/gecko/sync/repositories/android/AndroidBrowserRepositorySession.java:698]: You use new Integer(hashCode) in other places (I don't think that's actually necessary), but nit for consistency.

**Comment by liuche** [comment]
r+ from me.

**Comment by ncalexan** [comment]
This got merged under the tree name nalexander/bug-749729-shrink-guid-map.

**Pull request closed without merging**